services/wallet/router.go

```diff
@@ -0,0 +1,120 @@
+package wallet
+
+import (
+	"context"
+	"math"
+	"math/big"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/services/wallet/async"
+	"github.com/status-im/status-go/services/wallet/chain"
+)
+
+type SuggestedRoutes struct {
+	Networks []params.Network `json:"networks"`
+}
+
+func NewRouter(s *Service) *Router {
+	return &Router{s}
+}
+
+type Router struct {
+	s *Service
+}
+
+func (r *Router) suitableTokenExists(ctx context.Context, network *params.Network, tokens []*Token, account common.Address, amount float64, tokenSymbol string) (bool, error) {
+	for _, token := range tokens {
+		if token.Symbol != tokenSymbol {
+			continue
+		}
```

Comment:
What do you think of checking the token address instead of symbol?

Reply:
Ah no, this is supposed to be multi chain, right? so the addresses are gonna be different.

Reply:
correct, I need to use the symbol as key